packages/grid/_modules_/grid/hooks/features/selection/useGridSelection.ts

```diff
@@ -239,4 +211,23 @@ export const useGridSelection = (apiRef: GridApiRef): void => {
     });
     forceUpdate();
   }, [apiRef, setGridState, forceUpdate, isRowSelectable]);
+
+  React.useEffect(() => {
+    apiRef.current.registerControlState<GridSelectionModel, GridSelectionState>({
+      stateId: 'selectionModel',
+      propModel: props.selectionModel,
+      propOnChange: props.onSelectionModelChange,
+      stateSelector: (state) => state.selection,
+      mapStateToModel: (selectionState) => {
+        if (!selectionState) {
+          return undefined;
+        }
+        const model = Object.values(selectionState);
+        return model;
+      },
```

Comment:
Something seems off. Why don't we store the model as is, in the state, and have another place for transformed value?

Reply:
because sometimes we need to change the state to be more optimized for retrieving data

Reply:
I have just made the connection, it's related to https://github.com/mui-org/material-ui-x/pull/1864#discussion_r652278480. When implementing the control pattern. If we transform the controlled prop into a different shape and use it as a the source of truth, then we need a method to do the opposite *OR* to store the cache, somewhere else.

Reply:
Considering https://github.com/mui-org/material-ui-x/pull/1864#discussion_r654336662
I think we should stop transforming models in state and just transform them in selectors. So the state, would/should be the same as the props. And we can fix the === and remove mapModelToState as well

Reply:
All done as mentioned above so `mapStateToModel` is gone